File: Kattis/CPlusPlus/LastFactorialDigit.cpp
#include <iostream>

int calculateFactorial(int number); 

int main() {
	int i, j, cases, number, result, lastDigit;

	std::cin >> cases; 

	for (i = 0; i < cases; i++) {
	std::cin >> number;

	result = calculateFactorial(number);
	lastDigit = result % 10;
	std::cout << lastDigit; 
	}

	return 0;
}

int calculateFactorial(int number) {
	if (number == 1) {
		return 1;
	}
	else {
		return number * calculateFactorial(number - 1);
	}
}

File: Kattis/Java/LastFactorialDigit.java
package lastfactorialdigit;

import java.util.Scanner;

public class LastFactorialDigit {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int cases = input.nextInt();
         
        for (int i = 0; i < cases; i++) {
            int number = input.nextInt();
            
            int result = calculateFactorial(number);
            int lastDigit = result % 10;
            System.out.println(lastDigit);
        }
    }

    private static int calculateFactorial(int number) {
        if (number == 1) {
            return 1;
        } else {
            return number * calculateFactorial(number - 1);
        }
    }
}